File: Python/Owca.py
from Zwierze import Zwierze
from Organizm import Organizm

class Owca(Zwierze):
    def __init__(self, x, y, dz, swiat):
        super().__init__(x, y, dz, 4, 4, swiat.IDOWCA, swiat)

    def rysowanie(self):
        return "#ffffff"

    def nazwa(self):
        return "Owca"

    def stworz_nowe_zwierze(self, x, y, sw):
        org = Owca(x, y, True, sw)

File: C++/Owca.cpp
#include <iostream>
#include <windows.h>
#include "Owca.h"

Owca::Owca(int x, int y, Swiat* swiat) : Zwierze(x, y, 4, 4, IDOWCA, swiat)
{
}

Owca::Owca(int x, int y, bool dz, Swiat* swiat) : Zwierze(x, y, dz, 4, 4, IDOWCA, swiat)
{
}

void Owca::rysowanie()
{
	HANDLE col;
	col = GetStdHandle(STD_OUTPUT_HANDLE);
	SetConsoleTextAttribute(col, KOLOROWCA);
	std::cout << "O";
}

std::string Owca::nazwa()
{
	return "O";
}

void Owca::stworz_nowe_zwierze(int x, int y, Swiat* sw)
{
	Organizm* org = new Owca(x, y, true, sw);
}


Owca::~Owca()
{
}